spec/visual/samples/markdown

```diff
@@ -1042,6 +1042,12 @@ extensions:
 
 ``` ruby
 puts @foo
+
+def bar
+  puts <<-EOS
+    :foobar
+  EOS
+end
```

Comment:
unless `reset!`, NoMethodError raised.

```
  1) Error:
Rouge::Lexers::Rouge::Lexers::Markdown#test_0002_lexes the sample without throwing:
NoMethodError: undefined method `<<' for nil:NilClass
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/lexers/ruby.rb:234:in `block (2 levels) in <class:Ruby>'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:305:in `instance_exec'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:305:in `block in step'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:287:in `each'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:287:in `step'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:290:in `block in step'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:287:in `each'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:287:in `step'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:268:in `stream_tokens'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/lexer.rb:388:in `lex'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:365:in `delegate'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/lexers/markdown.rb:43:in `block (2 levels) in <class:Markdown>'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:305:in `instance_exec'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:305:in `block in step'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:287:in `each'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:287:in `step'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/regex_lexer.rb:268:in `stream_tokens'
    /Users/sei/src/github.com/yasslab/rouge/lib/rouge/lexer.rb:388:in `lex'
    /Users/sei/src/github.com/yasslab/rouge/spec/lexers_spec.rb:19:in `each'
    /Users/sei/src/github.com/yasslab/rouge/spec/lexers_spec.rb:19:in `to_a'
    /Users/sei/src/github.com/yasslab/rouge/spec/lexers_spec.rb:19:in `block (4 levels) in <top (required)>'
```



Ruby lexer initialize its state in `start` block,
https://github.com/jneen/rouge/blob/9083365cc92c324f458f46f249cd27b778bdccaf/lib/rouge/lexers/ruby.rb#L145

In README:

> this is run whenever a fresh lex is started

So I add `reset!` to markdown lexer